### PR TITLE
Use finch for sketching instead of mash

### PR DESCRIPTION
**Pull request by bovee**
In addition to a slight speed bump, Finch has some extra quality filtering (specifically for adapters and errors) that helps when handling raw FASTQ data. Note that this still uses mash for downstream distance calculations.

I'm not exactly sure the project format here, so please let me know if there's anything in this PR that needs to be changed or if more needs to added to get everything to run.

Thanks!
Roderick

### CODE REVIEW

File: snakemake_files/finch_sketch.smk [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    output:
+        sketch = config['mash_sketch']['sketch'],
+    threads: 1
+    params:
+        exc=config['finch']['exc']['filepath'],
+    resources:
+        time = 1,
+        n_gb_ram = 10
+    run:
+        cmd = '{params.exc} sketch --no-strict --seed 42 --n-hashes 10000000 --binary-format -o {output.sketch} {input.reads1}'
```
Comment (dcdanko) [inline snakemake_files/finch_sketch.smk]: parameters here should be sourced from the config

**Comment by dcdanko** [comment]
@bovee Looks like a good addition. Just looking for a minor edit thanks

**Comment by bovee** [comment]
@dcdanko Thanks! I think this commit addresses the parameter sourcing, but please let me know if those parameters aren't in the right place/format.

**Comment by dcdanko** [comment]
Yes looks good.

I'm going to hold off on the merge until the next version though.

**Comment by bovee** [comment]
Awesome, thanks!

**Comment by dcdanko** [comment]
@bovee Could you provide a link to this software? I'm having trouble finding it.

**Comment by bovee** [comment]
@dcdanko Not sure if you found this or still need it, but for a link to Finch: you can link to either the main repository at https://github.com/onecodex/finch-rs or the corresponding paper at http://joss.theoj.org/papers/10.21105/joss.00505

**Comment by dcdanko** [comment]
I did find it earlier but thank you. I forgot to check the documentation 🙄

**Pull request merged**